Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+    for (std::size_t i=0; i<n_nodes; ++i)
+        name_id_map[_id_nodename_map[i]] = i;
+    in.clear();
+    in.seekg(0, in.beg);
+
+    std::vector< MeshLib::Element* > elements;
+    std::vector<double> max_depth;
+    max_depth.resize(n_nodes);
+    std::size_t const n_types = 3;
+    std::array< std::size_t, n_types> n_elem_types;
+    while ( getline(in, line) )
+    {
+        if (line == "[RAINGAGES]")
+        {
+            if (!_rain_gauges.empty())
+                addRainGaugeTimeSeriesLocations(in);
+        }
+        if (line == "[SUBCATCHMENTS]")
+        {
+            INFO ("Reading subcatchment information...");
+            if (!readSubcatchments(in, name_id_map))
+                return false;
+        }
+        if (line == "[SUBAREAS]")
+        {
+            // more subcatchment variables, not yet implemented
+        }
+        if (line == "[INFILTRATION]")
+        {
+            // more subcatchment variables, not yet implemented
+        }
+        if (line == "[JUNCTIONS]")
+        {
+            INFO ("Reading junctions...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, true))
+                return false;
+        }
+        else if (line == "[OUTFALLS]")
+        {
+            INFO ("Reading outfalls...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, false))
+                return false;
+        }
+        else if (line == "[STORAGE]")
+        {
+            INFO ("Reading storages...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, true))
+                return false;
+        }
+        else if (line == "[CONDUITS]")
+        {
+            INFO ("Reading conduits...");
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[0] = elements.size();
+        }
+        else if (line == "[PUMPS]")
+        {
+            INFO ("Reading pumps...");
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[1] = elements.size();
+        }
+        else if (line == "[WEIRS]")
+        {
+            INFO ("Reading weirs...")
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[2] = elements.size();
+        }
+        else if (line == "[POLLUTANTS]")
+        {
+            if (!readPollutants(in))
+                return false;
+        }
+        if (line == "[Polygons]")
+        {
+            INFO ("Reading subcatchments...");
+            std::vector<GeoLib::Polyline*> lines;
+            std::vector<std::string> line_names;
+            std::vector<std::string> tmp_names; // polygon points are nameless but the method requires a vector
+            if (!readPolygons(in, lines, line_names, _subcatchment_points, tmp_names))
+                return false;
+
+            if (!matchSubcatchmentsWithOutlines(lines, line_names))
```

Comment:
Maybe rename the method to `matchSubcatchmentsToOutlines` the old name can be easiyl misread as `matchSubcatchmentsWithoutLines`